### ISSUE TITLE
`preserveModules` rollup support?

### ISSUE THREAD

**Issue by vitorbal** [preconstruct/preconstruct#354]
Hi all! Apologies if this was already discussed in another issue, I did some searching and couldn't find anything.
Have you considered adding support for rollup's [preserveModules](https://rollupjs.org/guide/en/#outputpreservemodules) flag?

This way, we could configure preconstruct to create a build that doesn't concatenate everything into a single file – which is required to get webpack to tree-shake at the module-level properly (see [this](https://github.com/webpack/webpack/issues/9337#issuecomment-522668890)] and [this](https://github.com/webpack/webpack/issues/9337#issuecomment-507179765)) for more context.

**Comment by mitchellhamilton** [comment]
I'm happy to accept a PR adding support for it with an experimental flag if it's not too complex

**Comment by daphnesmit** [comment]
Indeed! I changed my rollup setup to use preconstruct and my bundle got 60kb gzipped bigger then when I used custom rollup with preserveModules!
